devices/flash-flexspi/fspi/fspi.c

```diff
@@ -86,11 +81,41 @@ __attribute__((section(".noxip"))) void flexspi_lutUpdate(flexspi_t *fspi, u32 i
 }
 
 
+__attribute__((section(".noxip"))) void flexspi_lutUpdateEntries(flexspi_t *fspi, u32 index, const u32 *lutTable[], size_t elems, size_t count)
+{
+	u32 lutCopy[64];
+	const u32 *src;
+	unsigned int i, j;
+
+	for (i = 0; i < elems; ++i) {
+		src = *(lutTable++);
+		for (j = 0; j < count; ++j) {
+			lutCopy[i * count + j] = ((src != NULL) ? src[j] : 0);
+		}
+	}
+
+	flexspi_lutSet(fspi, index, lutCopy, elems * count);
+}
+
+
+__attribute__((section(".noxip"))) void flexspi_lutUpdate(flexspi_t *fspi, u32 index, const u32 *lutTable, size_t count)
+{
+	u32 lutCopy[64];
+	unsigned int i;
+
+	for (i = 0; i < count; ++i) {
+		lutCopy[i] = *lutTable++;
+	}
```

Comment:
why is this copy needed?

Reply:
Initially, the LUTs are on the flash from which we XIP, the flash read during XIP while updating the LUT entries (XIP read entry) is a guaranteed bus error. Function is run in ram (`.noxip` section) it's safe.